Library/Formula/crystal.rb

```diff
@@ -0,0 +1,22 @@
+class Crystal < Formula
+  desc "Code generator for every language, library and framework"
+  homepage "http://crystal.sh"
+  url "https://github.com/crystal/crystal/archive/v0.5.1.tar.gz"
+  sha256 "6c022ad59a10cec58856dba858a030b745333909108c08bd326c20ac0cb144fb"
+
+  depends_on "node"
+
+  def install
+    libexec.install Dir["*"]
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  def post_install
+    cd libexec
+    system "npm", "install"
```

Comment:
We should also move this back to `def install` so the `npm install` stage is included in the bottle. If we add `npm` to the `$PATH` with the above instruction, we can stop heavily leaning on `post_install` for Node stuff and deliver a slightly smoother UX.

`ENV["HOME"] = ".brew_home"` or similar needs adding as well, otherwise we'll have sandbox or leakage issues.